core/domain/customization_args_util.py

```diff
@@ -123,3 +131,217 @@ def validate_customization_args_and_values(
+        conversion_fn: function. The function to be used for converting the
+            HTML. It is passed the obj_type, customization argument value,
+            content_id prefix generated from schema, and the customization
+            argument name, if availible.
+    """
+    for ca_spec in ca_specs:
+        ca_spec_name = ca_spec.name
+        if ca_spec_name in ca_values:
+            find_subtitled(
+                ca_values[ca_spec_name],
+                ca_spec.to_dict(),
+                conversion_fn)
+
+
+def get_all_content_ids_in_cust_args(ca_values, ca_specs):
+    """Extracts all content_id's from ca_values, using ca_specs to find
+    SubtitledHtml or SubtitledUnicode locations.
+
+    Args:
+        ca_values: dict. The customization dict. The keys are names
+            of customization_args and the values are dicts with a
+            single key, 'value', whose corresponding value is the value of
+            the customization arg.
+        ca_specs: list(CustomizationArgSpec). List of spec dictionaries.
+
+    Returns:
+        list(str). List of all content_id's in customization arguments.
+    """
+    content_ids = []
+    def dummy_conversion(
+            ca_value, unused_obj_type, unused_content_id, unused_in_list):
+        """Conversion function used to extract all content_ids. Customization
+        argument is returned unmodified.
+
+        Args:
+            unused_obj_type: str. Indicates the obj_type found in
+                the customization arguments schema.
+            ca_value: dict. Dictionary of key 'value' to
+                original value of customization argument.
+            unused_content_id: str. The content_id generated from
+                traversing the customization argument spec.
+            unused_in_list: bool. Indicates if the content_id requires a
+                content index.
+
+        Returns:
+            str. The unmodified customization argument value.
+        """
+        if 'content_id' in ca_value:
+            content_ids.append(ca_value['content_id'])
+
+        return ca_value
+
+    for ca_spec in ca_specs:
+        ca_spec_name = ca_spec.name
+        if ca_spec_name in ca_values:
+            find_subtitled(
+                ca_values[ca_spec_name],
+                ca_spec.to_dict(),
+                dummy_conversion)
+
+    return content_ids
+
+
+def get_all_html_in_cust_args(ca_values, ca_specs):
+    """Extracts all html from ca_values, using ca_specs to find
+    SubtitledHtml locations.
+
+    Args:
+        ca_values: dict. The customization dict. The keys are names
+            of customization_args and the values are dicts with a
+            single key, 'value', whose corresponding value is the value of
+            the customization arg.
+        ca_specs: list(CustomizationArgSpec). List of spec dictionaries.
+
+    Returns:
+        list(str). List of all html in customization arguments.
+    """
+    html = []
+
+    def dummy_conversion(
```

Comment:
Please make sure that all your functions start with verbs.

Reply:
Done